File: 1894-merge-strings-alternately/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    const word1Length = word1.length; 
    const word2Length = word2.length; 
    let mergedString = ''
    if (word1Length === word2Length) {
        for(let i = 0; i < word1Length; i++) {
            mergedString += `${word1[i]}${word2[i]}`
        }
    } else if (word1Length > word2Length) {
        for (let i = 0; i < word2Length; i++) {
             mergedString += `${word1[i]}${word2[i]}`
        }
        for (let i = word2Length; i < word1Length; i++) {
            mergedString += word1[i]
        }
    } else {
        for (let i = 0; i < word1Length; i++) {
             mergedString += `${word1[i]}${word2[i]}`
        }
        for (let i = word1Length; i < word2Length; i++) {
            mergedString += word2[i]
        }
    }
    return mergedString; 
};

File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        merged_string = ''

        if len(word1) < len(word2):
            for i in range(0, len(word1), 1):
                merged_string = merged_string + word1[i] + word2[i]
            
            merged_string = merged_string + word2[-(len(word2)-len(word1)):]

        elif len(word2) < len(word1):
            for i in range(0, len(word2), 1):
                merged_string = merged_string + word1[i] + word2[i]
            
            merged_string = merged_string + word1[-(len(word1)-len(word2)):]
        
        else: 
            for i in range(0, len(word2), 1):
                merged_string = merged_string + word1[i] + word2[i]

        return merged_string